### PR TITLE
Add adjust angle vector ROS server/client

### PR DESCRIPTION
**Pull request by 708yamaguchi**
Usage

Command
```python
ri.angle_vector([0,0,0,0,0,0,0,0,0,0], 5)
ri.wait_interpolation(adjust_angle_vector=True, error_threshold=5)
```

Log
```
Servo [4] error [6.][deg] exceeds threshold [5.][deg].
Stop motion by overriding reference angle vector with current angle vector.
[WARN] [1724857818.102982]: Stop motion by adjust angle vector.
```

### CODE REVIEW

File: ros/kxr_controller/python/kxr_controller/kxr_interface.py [hunk 1]
```diff
@@ -90,6 +102,49 @@ def servo_off(self, joint_names=None):
...
+            adjust_angle_vector=False,
+            joint_names=None,
+            error_threshold=np.deg2rad(5),
+            **kwargs):
+        if adjust_angle_vector is False:
+            return super(KXRROSRobotInterface, self).wait_interpolation(
+                *args, timeout=timeout, **kwargs)
+        if timeout == 0:
+            timeout = float('inf')
+        start_time = time.time()
```
Comment (iory) [inline ros/kxr_controller/python/kxr_controller/kxr_interface.py:133]: We should use rospy.Time()

File: ros/kxr_controller/action/AdjustAngleVector.action [hunk 2]
```diff
@@ -0,0 +1,4 @@
+string[] joint_names
+float32[] error_threshold
+---
+---
```
Comment (iory) [inline ros/kxr_controller/action/AdjustAngleVector.action:4]: Please add end line

File: ros/kxr_controller/python/kxr_controller/kxr_interface.py [hunk 3]
```diff
@@ -90,6 +102,49 @@ def servo_off(self, joint_names=None):
...
+        # When all controllers finish interpolation or timeout,
+        # return from this function
+        while True:
```
Comment (iory) [inline ros/kxr_controller/python/kxr_controller/kxr_interface.py:136]: while rospy.is_shutdown() is better for KeyboardInterrupt

**Review by iory — CHANGES_REQUESTED** [review]

File: ros/kxr_controller/python/kxr_controller/kxr_interface.py (hunk 1 above)
Comment (708yamaguchi) [reply ros/kxr_controller/python/kxr_controller/kxr_interface.py]: I add commit [`7c5e625` (#77)](https://github.com/iory/rcb4/pull/77/commits/7c5e62515960887b6613536d55167a203dffb7d6)

File: ros/kxr_controller/action/AdjustAngleVector.action (hunk 2 above)
Comment (708yamaguchi) [reply ros/kxr_controller/action/AdjustAngleVector.action]: I add commit [`54f122a` (#77)](https://github.com/iory/rcb4/pull/77/commits/54f122ad6d5ce7c5892c5a1b5a3381b35db94afc)

**Comment by 708yamaguchi** [comment]
Thank you for your review.
I pushed some commits.

**Comment by 708yamaguchi** [comment]
I rebased the latest origin/master and tested this PR again.

**Pull request merged**